lib/active_merchant/errors.rb

```diff
@@ -23,6 +23,12 @@ def initialize(response, message = nil)
     end
 
     def to_s
+      if response.kind_of?(String)
+        return response if response.start_with?('Failed with')
```

Comment:
As-is, an arbitrary `String` response that doesn't start with `Failed with` would say `ActiveMerchant::OAuthResponseError: Failed with`. Is that desirable?

Reply:
Yes exactly. I add this check because [some gateways like Alelo check access_token method in `commit`](https://github.com/activemerchant/active_merchant/pull/4869/files#diff-9ab1fec4c8a427ccb1bbff06670245bf4f9adf54e536926b01fc8aeba6151634R221) so it would return `ActiveMerchant::OAuthResponseError: Failed with Failed with`.

Reply:
But wouldn't returning the the `response` string in that case be more informative?

Reply:
In the case I described the OAuthResponseError or maybe I'm not understanding what you trying to say

Reply:
I mean that error strings like `An error happened` would return `ActiveMerchant::OAuthResponseError: Failed with` and not fall to the general case `ActiveMerchant::OAuthResponseError: Failed with An error happened`. So I'm wondering if the `if response.start_with?('Failed with')` check is necessary, when it could read as `ActiveMerchant::OAuthResponseError: An error happened` without that check.

Reply:
Oh, if response is a string then it would be a `ActiveMerchant::OAuthResponseError`. If response was `ActiveMerchant::OAuthResponseError` the it would fall into the second statement, `elsif response.respond_to?(:message)`.